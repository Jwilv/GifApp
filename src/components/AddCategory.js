import React, { useState } from 'react'    
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleInpuntChange = element => { setInputValue(element.target.value) }

    const handleSubmit = element => {
        element.preventDefault();
        //console.log('salio bien')
        if(inputValue.trim().length > 0){
            setCategories(categorie =>[inputValue,...categorie]);
            setInputValue('')
        }
        
    }
    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={handleInpuntChange}
            />
        </form>
    )

}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}