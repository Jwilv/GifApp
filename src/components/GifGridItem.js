import React from 'react'

export const GifGridItem = ({title, url }) => {

    console.log({url})

    return (
        <div className='card animate__animated animate__backInLeft'>
        <img src={url} alt={title}/>
        <p>{title}</p>
        </div>
    )
}
