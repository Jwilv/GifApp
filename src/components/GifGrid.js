import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h2 className='animate__animated animate__backInLeft'>{category}</h2>
      <div className='card-grid'>
        {loading && <p>loading...</p>}
        {
          images.map(img => (
            <GifGridItem
              key={img.id}
              {...img}
            />
          ))
        }
      </div>
    </>
  )
}
