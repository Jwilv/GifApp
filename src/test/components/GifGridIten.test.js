import React from 'react'
import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem'

describe('pruebas en <GifGridItem />', () => {
    test('debe mostrar el commponente correctamente', () => {
        const wraper = shallow(<GifGridItem />)

        expect(wraper).toMatchSnapshot()

    })

})