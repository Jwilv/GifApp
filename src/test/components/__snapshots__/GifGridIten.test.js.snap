// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifGridItem /> debe mostrar el commponente correctamente 1`] = `
<div
  className="card animate__animated animate__backInLeft"
>
  <img />
  <p />
</div>
`;
